src/component/Simulator/Simulator.tsx

```diff
@@ -220,82 +220,28 @@ export class Simulator extends React.Component<SimulatorProps, SimulatorState> {
         );
     }
 
-    private getAssetServer(): any {
-        return {
-            type_urls: {
-                flow: 'http://localhost:9000/flow/{uuid}/',
-                field_set: 'http://localhost:9000/fields/',
-                channel_set: 'http://localhost:9000/channels/',
-                group_set: 'http://localhost:9000/groups/'
-            }
-        };
-    }
-
-    private getAssets(): Asset[] {
+    private getCurrentDefinition(): FlowDefinition {
         const renderNodes = getOrderedNodes(this.props.nodes);
         const nodes: FlowNode[] = [];
         renderNodes.map((renderNode: RenderNode) => {
             nodes.push(renderNode.node);
         });
-
         this.props.definition.nodes = nodes;
-
-        return [
-            {
-                type: 'flow',
-                url: 'http://localhost:9000/flow/' + this.props.definition.uuid + '/',
-                content: this.props.definition
-            },
-            /*{
-                type: 'field_set',
-                url: 'http://localhost:9000/fields/',
-                content: this.props.contactFields.map((field: SearchResult) => {
-                    return { key: field.id, name: field.name, value_type: 'text' };
-                })
-            },
-            {
-                type: 'group_set',
-                url: 'http://localhost:9000/groups/',
-                content: this.props.groups.map((group: SearchResult) => {
-                    return { uuid: group.id, name: group.name };
-                })
-            },*/
-            {
-                type: 'channel_set',
-                url: 'http://localhost:9000/channels/',
-                content: [
-                    {
-                        uuid: '57f1078f-88aa-46f4-a59a-948a5739c03d',
-                        name: 'Simulator',
-                        address: '+12345671111',
-                        schemes: ['tel'],
-                        roles: ['send', 'receive']
-                    }
-                ]
-            }
-        ];
+        return this.props.definition;
```

Comment:
don't know if it matters that we're reassigning props or if it could have any negative consequences but we could also return a new object, e.g. 

```
private getCurrentDefinition(): FlowDefinition {
    const renderNodes = getOrderedNodes(this.props.nodes);
    const nodes = renderNodes.map(({ node }: RenderNode) => node);
    return {
        ...this.props.definition,
        nodes
    };
}
```

Reply:
ya, this definitely felt hacky. New object coming up.